test/Operators/finitedifference/stencil_operations.jl

```diff
@@ -0,0 +1,212 @@
+
+# op1.(a)[i] = ∑_j C[i, j] * a[j]                                            ==>
+# op1.(a1 .* a0)[i] =
+#   ∑_j C[i, j] * a1[j] * a0[j] =
+#   ∑_j stencil_op1.(a1)[i, j] * a0[j] =
+#   apply.(stencil_op1.(a1), a0)[i]                                          ==>
+# op1.(a1 .* a0) = apply.(stencil_op1.(a1), a0)
+
+# op1.(a)[i] = ∑_j C1[i, j] * a[j] and op2.(a)[i] = ∑_k C2[i, k] * a[k]      ==>
+# op2.(a2 .* op1.(a1 .* a0))[i] =
+#   ∑_k C2[i, k] * a2[k] * op1.(a1 .* a0)[k] =
+#   ∑_k C2[i, k] * a2[k] * (∑_j C1[k, j] * a1[j] * a0[j]) =
+#   ∑_j (∑_k C2[i, k] * a2[k] * C1[k, j] * a1[j]) * a0[j] =
+#   ∑_j (∑_k stencil_op2.(a2)[i, k] * stencil_op1.(a1)[k, j]) * a0[j] =
+#   ∑_j compose.(stencil_op2.(a2), stencil_op1.(a1))[i, j] * a0[j] =
+#   apply.(compose.(stencil_op2.(a2), stencil_op1.(a1)), a0)[i]              ==>
+# op2.(a2 .* op1.(a1 .* a0)) =
+#   apply.(compose.(stencil_op2.(a2), stencil_op1.(a1)), a0)
+
+struct CurriedTwoArgOperator{O, A}
+    op::O
+    arg2::A
+end
+
+Base.Broadcast.broadcasted(op::CurriedTwoArgOperator, arg1) =
+    Base.Broadcast.broadcasted(op.op, arg1, op.arg2)
+
+Operators.Operator2Stencil(op::CurriedTwoArgOperator) =
+    CurriedTwoArgOperator(Operators.Operator2Stencil(op.op), op.arg2)
+
+@testset "Pointwise Stencil Construction/Composition/Application" begin
+    seed!(1) # ensures reproducibility
+
+    for FT in (Float32, Float64)
+        radius = FT(1e7)
+        zmax = FT(1e4)
+        helem = npoly = 2
+        velem = 4
+
+        hdomain = Domains.SphereDomain(radius)
+        hmesh = Meshes.EquiangularCubedSphere(hdomain, helem)
+        htopology = Topologies.Topology2D(hmesh)
+        quad = Spaces.Quadratures.GLL{npoly + 1}()
+        hspace = Spaces.SpectralElementSpace2D(htopology, quad)
+
+        vdomain = Domains.IntervalDomain(
+            Geometry.ZPoint{FT}(zero(FT)),
+            Geometry.ZPoint{FT}(zmax);
+            boundary_tags = (:bottom, :top),
+        )
+        vmesh = Meshes.IntervalMesh(vdomain, nelems = velem)
+        vspace = Spaces.CenterFiniteDifferenceSpace(vmesh)
+
+        # TODO: Replace this with a space that includes topography.
+        center_space = Spaces.ExtrudedFiniteDifferenceSpace(hspace, vspace)
+        center_coords = Fields.coordinate_field(center_space)
+        face_coords = Fields.coordinate_field(
+            Spaces.FaceExtrudedFiniteDifferenceSpace(center_space),
+        )
+
+        # We can't use non-zero non-extrapolation BCs because Operator2Stencil
+        # does not account for them (it only handles linear transformations).
+        zero_scalar = Operators.SetValue(zero(FT))
+        zero_vector = Operators.SetValue(Geometry.Covariant3Vector(zero(FT)))
+        zero_grad = Operators.SetGradient(Geometry.Covariant3Vector(zero(FT)))
+        zero_div = Operators.SetDivergence(zero(FT))
+        zero_curl = Operators.SetCurl(Geometry.Contravariant3Vector(zero(FT)))
+        extrap = Operators.Extrapolate()
```

Comment:
Can we please add tests with non-zero BCs for `SetValue`, `SetGradient` and `SetDivergence`?

Reply:
That would correspond to an affine transformation (rather than a linear transformation), which I cannot currently represent with my stencil data structure. I do plan to add that capability in the future, though.

Reply:
To clarify, I could add non-zero BCs to the unit tests, and they would run, but the tests would not pass because the "linearized" expressions would not be equal to the "exact" expressions at the boundaries.